### PR TITLE
[APID-159] Upgrade rocksteady to Ruby 3.1.x

### PR DESCRIPTION
**Pull request by PatoSoft**
- [Upgrade rocksteady to ruby 3.1.x](https://github.com/PowerRhino/rocksteady/commit/2d0b8ad37246654c56002abfad2b893372feb334)
- [Remove future deprecation warnings](https://github.com/PowerRhino/rocksteady/commit/9b71d96d2e428990a156e1c6e98a32ccb88cdd69)
- [Bump nomad version](https://github.com/PowerRhino/rocksteady/pull/97/commits/26799a6d21fee24425d26a55916078299ddb6770)

### CODE REVIEW

**Review by mowat27 — COMMENTED** [review]
Looks good. One small query about a test - although I don't feel strongly about it.  Also one test in the test suite takes over a minute to run.  Have you seen this:

```
  AppsController#index when HTML format renders the index template
    78.26 seconds ./spec/controllers/apps_controller_spec.rb:20
```

File: spec/services/graylog_api/client_spec.rb [hunk 1]
```diff
@@ -114,7 +114,7 @@ def stub_post_with(status:, body:)
     end
 
     it 'returns the full error message' do
-      expect(client.put('/endpoint', pay: 'load').body[:stack_trace]).to include('1: from')
+      expect(client.put('/endpoint', pay: 'load').body[:stack_trace]).to include('raise_timeout_error')
```
[the remark this answers is not in the document]
Comment (PatoSoft) [reply spec/services/graylog_api/client_spec.rb]: Thank you for the suggestion @mowat27 I changed it here 9c62b1bf88f743bb13556a870f35ebb18a66d5ac

**Review by ttr3dp — APPROVED** [review]
🍰 LGTM!

+ if you could just bump the nomad version to `1.3.1`, since that's what we're using in our envs. (https://github.com/PowerRhino/rocksteady/blob/master/docker-dev/nomad/Dockerfile#L6)

@mowat27 as for that long running spec. Bump actually reduced the time. It took 142.45 seconds on `master`

**Comment by PatoSoft** [comment]
> 🍰 LGTM!
> 
> * if you could just bump the nomad version to `1.3.1`, since that's what we're using in our envs. (https://github.com/PowerRhino/rocksteady/blob/apid-159/upgrade-rocksteady-to-ruby-3.1.x/docker-dev/nomad/Dockerfile#L6)
> 
> @mowat27 as for that long running spec. Bump actually reduced the time. It took ~140 seconds on `master`

Done 26799a6d21fee24425d26a55916078299ddb6770 @ttr3dp

**Pull request merged**